http-server/src/module.rs

```diff
@@ -91,10 +97,16 @@ impl<Context> RpcContextModule<Context> {
 		self.module.methods.insert(
 			method_name,
 			Box::new(move |id, params, tx, _| {
-				let result = callback(params, &*ctx)?;
-
-				send_response(id, tx, result);
-
+				match callback(params, &*ctx) {
+					Ok(res) => send_response(id, tx, res),
+					Err(ServerCallError::InvalidParams(_)) => {
+						send_error(id, tx, JsonRpcErrorCode::InvalidParams.into())
+					}
+					Err(ServerCallError::ContextFailed(err)) => {
+						let err = JsonRpcErrorObject { code: 1.into(), message: &err.to_string(), data: None };
```

Comment:
hardcoded to `1` :cold_face:

Reply:
I think it ends up [here](https://github.com/paritytech/jsonrpsee/blob/c106710a2f7c14c2deacd549bcd5d7beccf8d121/types/src/v2/error.rs#L136) and that the number doesn't matter?
Still, not sure I follow why not use `JsonRpcErrorCode::ServerError(1)` explicitly here? And what the plan is for the `ServerError` code is.

Reply:
yepp, I think we could dedicate/document that an error code for this but `-32000 to -32099 | Server error | Reserved for implementation-defined server-errors` seems better for this